File: src/tauri-adapter/stats.ts
import { invoke } from "@tauri-apps/api/core";

export interface TrendDataPoint {
  date: string;
  income: number;
  expense: number;
}

export interface CategoryBreakdown {
  category_name: string;
  amount: number;
  percent: number;
}

export interface AssetData {
  account_type: string;
  balance: number;
}

export const StatsAPI = {
  getMonthlyTrend: async (ledgerId: string, startTs: number, endTs: number): Promise<TrendDataPoint[]> => {
    return invoke("get_monthly_trend", { ledgerId, startTs, endTs });
  },
  
  getCategoryBreakdown: async (ledgerId: string, startTs: number, endTs: number, txType: "expense" | "income"): Promise<CategoryBreakdown[]> => {
    return invoke("get_category_breakdown", { ledgerId, startTs, endTs, txType });
  },
  
  getAssetsOverview: async (): Promise<AssetData[]> => {
    return invoke("get_assets_overview");
  }
};

File: src-tauri/src/commands/stats.rs
use tauri::State;
use crate::adapters::sqlite_store::SqliteStore;
use crate::error::AppError;

#[tauri::command]
pub fn get_monthly_trend(store: State<'_, SqliteStore>, ledger_id: String, start_ts: i64, end_ts: i64) -> Result<Vec<crate::core::stats_engine::TrendDataPoint>, AppError> {
    let engine = crate::core::stats_engine::StatsEngine::new(store.get_conn());
    engine.get_monthly_trend(&ledger_id, start_ts, end_ts)
}

#[tauri::command]
pub fn get_category_breakdown(store: State<'_, SqliteStore>, ledger_id: String, start_ts: i64, end_ts: i64, tx_type: String) -> Result<Vec<crate::core::stats_engine::CategoryBreakdown>, AppError> {
    let engine = crate::core::stats_engine::StatsEngine::new(store.get_conn());
    engine.get_category_breakdown(&ledger_id, start_ts, end_ts, &tx_type)
}

#[tauri::command]
pub fn get_assets_overview(store: State<'_, SqliteStore>) -> Result<Vec<crate::core::stats_engine::AssetData>, AppError> {
    let engine = crate::core::stats_engine::StatsEngine::new(store.get_conn());
    engine.get_assets_overview()
}
